File: api/javer/src/main/java/com/uuhnaut69/javer/model/Dish.java
package com.uuhnaut69.javer.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ManyToOne;
import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "dishes")
public class Dish extends BaseEntity {

  @Column(columnDefinition = "text", nullable = false)
  private String name;

  @Column(nullable = false)
  private BigDecimal price;

  @JsonIgnore
  @ManyToOne(fetch = FetchType.LAZY)
  private Restaurant restaurant;
}

File: api/kover/src/main/kotlin/com/uuhnaut69/kover/model/Dish.kt
package com.uuhnaut69.kover.model

import com.fasterxml.jackson.annotation.JsonIgnore
import java.math.BigDecimal
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.ManyToOne

@Entity(name = "dishes")
class Dish : BaseEntity() {

    @Column(columnDefinition = "text", nullable = false)
    var name: String? = null

    @Column(nullable = false)
    var price: BigDecimal? = null

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    var restaurant: Restaurant? = null

}